stripe-core/src/main/java/com/stripe/android/core/injection/InjectWithFallback.kt

```diff
@@ -31,3 +31,37 @@ fun <FallbackInitializeParam> Injectable<FallbackInitializeParam>.injectWithFall
         fallbackInitialize(fallbackInitializeParam)
     }
 }
+
+/**
+ * Try to use an [InjectorKey] to retrieve an [Injector] and inject, if no [Injector] is found,
+ * invoke [Injectable.fallbackInitialize] with [fallbackInitializeParam].
+ * Used by classes that are responsible for providing a [NonFallbackInjector] to other classes.
+ *
+ * @return The [NonFallbackInjector] used to inject the dependencies into this class.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun <FallbackInitializeParam> Injectable<FallbackInitializeParam, NonFallbackInjector>.injectWithFallback(
```

Comment:
`injectWithFallback` was created to encapsulate all injection logic, now that this method returns a injector, should we name it differently?

Reply:
This is still the method that does the injection, with fallback to `fallbackInitialize`. The only difference is that it also returns the injector. Do you have a better name?